packages/rpc/src/RpcSerialization.ts

```diff
@@ -73,6 +75,152 @@ export const ndjson: Effect.Effect<RpcSerialization["Type"]> = Effect.sync(() =>
   })
 })
 
+/**
+ * @since 1.0.0
+ * @category serialization
+ */
+export const jsonrpc: Effect.Effect<RpcSerialization["Type"]> = Effect.sync(() => {
+  const decoder = new TextDecoder()
+  return RpcSerialization.of({
+    contentType: "application/json-rpc",
+    supportsBigInt: false,
+    unsafeMake: () => ({
+      decode: (bytes) => {
+        const decoded: JsonRpcMessage = JSON.parse(typeof bytes === "string" ? bytes : decoder.decode(bytes))
+        let rpcMessage: RpcMessage.FromClientEncoded | RpcMessage.FromServerEncoded
+
+        if ("method" in decoded) {
+          rpcMessage = {
+            _tag: "Request",
+            id: String(decoded.id),
+            tag: decoded.method,
+            payload: decoded.params,
+            headers: decoded.headers ?? [],
+            traceId: decoded.traceId ?? "noop",
+            spanId: decoded.spanId ?? "noop",
+            sampled: decoded.sampled ?? false
+          }
+        } else if (decoded.error && decoded.error._tag === "Defect") {
+          rpcMessage = {
+            _tag: "Defect",
+            defect: decoded.error.data
+          }
+        } else {
+          rpcMessage = decoded.chunk === true ?
+            {
+              _tag: "Chunk",
+              requestId: String(decoded.id),
+              values: decoded.result as any
+            } :
+            {
+              _tag: "Exit",
+              requestId: String(decoded.id),
+              exit: decoded.error != null ?
+                {
+                  _tag: "Failure",
+                  cause: decoded.error._tag === "Cause" ?
+                    decoded.error.data as any :
+                    {
+                      _tag: "Die",
+                      defect: decoded.error
+                    }
+                } :
+                {
+                  _tag: "Success",
+                  value: decoded.result
+                }
+            }
+        }
+
+        return [rpcMessage]
+      },
+      encode: (response_) => {
+        const response = response_ as RpcMessage.FromServerEncoded | RpcMessage.FromClientEncoded
+
+        let jsonRpcMessage: JsonRpcMessage
+        if (response._tag === "Request") {
+          jsonRpcMessage = {
+            jsonrpc: "2.0",
+            method: response.tag,
+            params: response.payload,
```

Comment:
Generally for the RPC module, `payload` should be:

a) optional
b) not just `Schema.Struct`

E.g. JSON RPC commonly uses positional tuples for payloads and for some methods, the payload can also be optional entirely. In these cases, the `params` key should not be set (but most implementations will accept an empty array or empty object anyways).